addon-test-support/@ember/test-helpers/dom/-get-element.ts

```diff
@@ -1,6 +1,15 @@
 import getRootElement from './get-root-element';
 import Target, { isDocument, isElement } from './-target';
 
+function getElement<
+  K extends keyof (HTMLElementTagNameMap | SVGElementTagNameMap)
+>(target: K): (HTMLElementTagNameMap | SVGElementTagNameMap)[K] | null;
+function getElement<K extends keyof HTMLElementTagNameMap>(
+  target: K
+): HTMLElementTagNameMap[K] | null;
+function getElement<K extends keyof SVGElementTagNameMap>(
+  target: K
+): SVGElementTagNameMap[K] | null;
```

Comment:
These types started as copy-pasta from the built-in `querySelector` types, which look like this:

```ts
    /** Returns the first element that is a descendant of node that matches selectors. */
    querySelector<K extends keyof HTMLElementTagNameMap>(selectors: K): HTMLElementTagNameMap[K] | null;
    querySelector<K extends keyof SVGElementTagNameMap>(selectors: K): SVGElementTagNameMap[K] | null;
    querySelector<E extends Element = Element>(selectors: string): E | null;
```

...but as @chriskrycho pointed out [here](https://github.com/emberjs/ember-test-helpers/pull/1287#discussion_r1049755184), those types aren't that great.

Since we're diverging from them a bit by removing the generic from the `target: string` overload, I decided to also fix a type bug from those types, which is this:

`HTMLElementTagNameMap` and `SVGElementTagNameMap` have some overlap in keys, e.g. `a` is a valid tag in both. In the built-in `querySelector` types, the return type for `target: 'a'` would always be an `HTMLAnchorElement`, when the actual value might actually be a `SVGAElement`. So I added a third overload to account for that overlap. You can see the results in the type-tests.